packages/jest-runtime/src/index.ts

```diff
@@ -2231,8 +2231,30 @@ export default class Runtime {
     const jestObject: Jest = {
       advanceTimersByTime: (msToRun: number) =>
         _getFakeTimers().advanceTimersByTime(msToRun),
+      advanceTimersByTimeAsync: (msToRun: number): Promise<number> => {
+        const fakeTimers = _getFakeTimers();
+
+        if (fakeTimers === this._environment.fakeTimersModern) {
+          return fakeTimers.advanceTimersByTimeAsync(msToRun);
```

Comment:
I wonder if these needs an extra check, so we avoid a new #13822 with people using custom test environments with dated dependencies

Reply:
Do you mean adding the check `typeof this._moduleMocker.replaceProperty === 'function'` in the existing condition?

Reply:
no, checking that `typeof fakeTimers.advanceTimersByTimeAsync === 'function'`